presto-native-execution/presto_cpp/main/expression/tests/RowExpressionEvaluatorTest.cpp

```diff
@@ -0,0 +1,180 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "presto_cpp/main/expression/RowExpressionEvaluator.h"
+#include <folly/SocketAddress.h>
+#include <folly/Uri.h>
+#include <folly/executors/IOThreadPoolExecutor.h>
+#include <gtest/gtest.h>
+#include "presto_cpp/main/common/tests/test_json.h"
+#include "presto_cpp/main/http/tests/HttpTestBase.h"
+#include "velox/exec/OutputBufferManager.h"
+#include "velox/expression/RegisterSpecialForm.h"
+#include "velox/functions/prestosql/registration/RegistrationFunctions.h"
+#include "velox/parse/TypeResolver.h"
+#include "velox/vector/VectorStream.h"
+#include "velox/vector/tests/utils/VectorTestBase.h"
+
+using namespace facebook::presto;
+using namespace facebook::velox;
+
+namespace {
+std::string getDataPath(const std::string& fileName) {
+  std::string currentPath = fs::current_path().c_str();
+
+  if (boost::algorithm::ends_with(currentPath, "fbcode")) {
+    return currentPath +
+        "/github/presto-trunk/presto-native-execution/presto_cpp/main/types/tests/data/" +
+        fileName;
+  }
+
+  if (boost::algorithm::ends_with(currentPath, "fbsource")) {
+    return currentPath + "/third-party/presto_cpp/main/types/tests/data/" +
+        fileName;
+  }
+
+  // CLion runs the tests from cmake-build-release/ or cmake-build-debug/
+  // directory. Hard-coded json files are not copied there and test fails with
+  // file not found. Fixing the path so that we can trigger these tests from
+  // CLion.
+  boost::algorithm::replace_all(currentPath, "cmake-build-release/", "");
+  boost::algorithm::replace_all(currentPath, "cmake-build-debug/", "");
+
+  return currentPath + "/data/" + fileName;
+}
+} // namespace
+
+class RowExpressionEvaluatorTest
+    : public ::testing::Test,
+      public facebook::velox::test::VectorTestBase {
+ protected:
+  static void SetUpTestCase() {
+    memory::MemoryManager::testingSetInstance({});
+  }
+
+  void SetUp() override {
+    parse::registerTypeResolver();
+    functions::prestosql::registerAllScalarFunctions("presto.default.");
+    exec::registerFunctionCallToSpecialForms();
+
+    auto httpServer = std::make_unique<http::HttpServer>(
```

Comment:
You can test RowExpressionEvaluator without all this http stubing if abstracting as suggested. That is cleaner.

Reply:
Could you please share how to get rid of this http stubbing? I wasn't quite able to follow how it can be refactored.